### PR TITLE
Sed 1644 introduce versioning of entities

### PR DESCRIPTION
**Pull request by david-stephan**

### CODE REVIEW

File: step-controller/step-controller-base-plugins/src/test/java/step/plugins/parametermanager/ParameterManagerTest.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public void testPerf() throws ScriptException {
 		System.out.println("ms:"+(System.currentTimeMillis()-t1));
 		Assert.assertEquals(params.get("key1"),"value"+nIt);
 		
-		Assert.assertTrue((System.currentTimeMillis()-t1)<500);	
+		Assert.assertTrue((System.currentTimeMillis()-t1)<5000);
```
Comment (jeromecomte) [inline step-controller/step-controller-base-plugins/src/test/java/step/plugins/parametermanager/ParameterManagerTest.java:113]: The performance of the method getAllParameterValues is critical as it is executed for each execution start. Did the changes of this PR impact the performance of this method?

File: step-controller/step-controller-server/src/main/java/step/controller/services/entities/AbstractEntityServices.java [hunk 2]
```diff
@@ -197,4 +198,35 @@ protected <R> AsyncTaskStatus<R> scheduleAsyncTaskWithinSessionContext(AsyncTask
     public TableResponse<T> request(TableRequest request) throws TableServiceException {
         return tableService.request(entityName, request, getSession());
     }
+
+    @Operation(operationId = "get{Entity}History", description = "Retrieves entity's versioned")
```
Comment (jeromecomte) [inline step-controller/step-controller-server/src/main/java/step/controller/services/entities/AbstractEntityServices.java:202]: entity's versions.... or
Retrieves the versions of the entity with the given id...

File: step-controller/step-controller-server/src/main/java/step/core/Controller.java [hunk 3]
```diff
@@ -131,8 +131,10 @@ private void initContext() throws ClassNotFoundException, PluginManager.Builder.
 				collectionFactory.getCollection("executions", Execution.class));
 		context.setExecutionAccessor(executionAccessor);		
 		context.setExecutionManager(new ExecutionManagerImpl(executionAccessor));
-		
-		context.setPlanAccessor(new PlanAccessorImpl(collectionFactory.getCollection("plans", Plan.class)));
+
+		PlanAccessorImpl plans = new PlanAccessorImpl(collectionFactory.getCollection("plans", Plan.class));
```
Comment (jeromecomte) [inline step-controller/step-controller-server/src/main/java/step/core/Controller.java:135]: No change?

**Review by jeromecomte — DISMISSED** [review]
Couldn't find the place where the versioning is enabled. Assume this isn't possible in OS. See minor line comments

**Comment by david-stephan** [comment]
Correct this enabled in Ee

Le mar. 31 janv. 2023 à 16:50, Jérôme Comte ***@***.***> a
écrit :

File: step-controller/step-controller-server/src/main/java/step/core/Controller.java (hunk 3 above)
Comment (david-stephan) [reply step-controller/step-controller-server/src/main/java/step/core/Controller.java:135]: the versioning is enabled in an EE plugin

File: step-controller/step-controller-server/src/main/java/step/controller/services/entities/AbstractEntityServices.java (hunk 2 above)
Comment (david-stephan) [reply step-controller/step-controller-server/src/main/java/step/controller/services/entities/AbstractEntityServices.java:202]: changes made

File: step-controller/step-controller-base-plugins/src/test/java/step/plugins/parametermanager/ParameterManagerTest.java (hunk 1 above)
Comment (david-stephan) [reply step-controller/step-controller-base-plugins/src/test/java/step/plugins/parametermanager/ParameterManagerTest.java]: @jeromecomte : due to the importance of the tests, I switch it to use a mongo collection and the central DB, and compare the performance between master and the branch which are identical. Note that I add to increase the threshold to 2 seconds for mongo.

**Pull request merged**